src/com/github/mtdrewski/GRAPH_moment/controller/GraphDrawer.java

```diff
@@ -115,14 +115,22 @@ private void reDrawGraph(){
             VertexCircle vertexShape= vertexShapeFactory.get();
             vertexShape.setCenterX(v1.xPos());
             vertexShape.setCenterY(v1.yPos());
-            anchorPane.getChildren().add(vertexShape);
             vertexCircles.add(vertexShape);
+            anchorPane.getChildren().add(vertexShape);
         }
 
         for(Edge e1:graph.getEdges()){
-            EdgeLine edgeLine = edgeLineFactory.get();
-            edgeLine.setStartVertex(vertexCircles.get(e1.vert1().id()-1));
+
+            sourceVertex=vertexCircles.get(e1.vert1().id()-1);
```

Comment:
I didn't describe it anywhere so sorry for that but, sourceVertex is only to indicate a vertex from which there outgoes an edge which follows the mouse while being in "Edge Mode". So sourceVertex is only for purposes of Edge mode and you shouldn't use it for purpose of redrawing graph